pkg/controller/populators/clone-populator.go

```diff
@@ -81,6 +88,90 @@ type ClonePopulatorReconciler struct {
 	multiTokenValidator *cc.MultiTokenValidator
 }
 
+var supportedCloneSources = map[string]client.Object{
+	"VolumeSnapshot":        &snapshotv1.VolumeSnapshot{},
+	"PersistentVolumeClaim": &corev1.PersistentVolumeClaim{},
+}
+
+func isSourceReady(obj client.Object) bool {
+	switch obj := obj.(type) {
+	case *snapshotv1.VolumeSnapshot:
+		return cc.IsSnapshotReady(obj)
+	case *corev1.PersistentVolumeClaim:
+		return cc.IsBound(obj)
+	}
+	return false
+}
+
+func addCloneSourceWatches(mgr manager.Manager, c controller.Controller, log logr.Logger) error {
+	getKey := func(kind, namespace, name string) string {
+		return kind + "/" + namespace + "/" + name
+	}
+
+	if err := mgr.GetClient().List(context.TODO(), &snapshotv1.VolumeSnapshotList{}); err != nil {
```

Comment:
So I don't know if this is a real world issue, but, if there is not snapshot support in the cluster, you'd back out and not set up a PVC watch

Reply:
I considered it but I don't know how much I should worry about this case... Don't mind handling it if necessary but code might become more complex.

Reply:
Not so complex:
https://github.com/kubevirt/containerized-data-importer/blob/f4ddcc1748f8b2eeb14eb46e94b658a47e66445b/pkg/controller/dataimportcron-controller.go#L1199-L1207

Reply:
By complex I mean that it requires duplicating the watches and mappers instead of having a list of supported sources, but yeah I'll change it if we prefer to handle this error.

Reply:
ack, found a way to do this without duplicating the code, hope it's ok

Reply:
@akalenyu @arnongilboa gentle ping